webapp/contacts_reader.py

```diff
@@ -0,0 +1,113 @@
+from argparse import ArgumentParser, FileType
+from collections import OrderedDict
+import hashlib
+import os
+import sys
+from typing import Dict
+
+import anymarkup
+
+# thanks stackoverflow
+if __name__ == "__main__" and __package__ is None:
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from webapp.common import MaybeOrderedDict, to_xml, MISCUSER_SCHEMA_URL
+
+
+class User(object):
+    def __init__(self, id_, yaml_data):
+        self.id = id_
+        self.yaml_data = yaml_data
+
+    def get_tree(self, authorized=False, filters=None) -> MaybeOrderedDict:
+        tree = OrderedDict()
+        tree["FullName"] = self.yaml_data["FullName"]
+        tree["PhotoURL"] = self.yaml_data.get("PhotoURL", None)
+        tree["GravatarURL"] = self._get_gravatar_url(
+            self.yaml_data["ContactInformation"]["PrimaryEmail"])
+        tree["Profile"] = self.yaml_data.get("Profile", None)
+        if authorized:
+            tree["ContactInformation"] = self._expand_contact_info()
+        return tree
+
+    @property
+    def name(self):
+        return self.yaml_data["FullName"]
+
+    @property
+    def email(self):
+        return self.yaml_data["ContactInformation"]["PrimaryEmail"]
+
+    @property
+    def phone(self):
+        return self.yaml_data["ContactInformation"].get("PrimaryPhone", None)
+
+    @property
+    def sms_address(self):
+        return self.yaml_data["ContactInformation"].get("SMSAddress", None)
+
+    @property
+    def dns(self):
+        dns = self.yaml_data["ContactInformation"].get("DNs", None)
+        return dns
```

Comment:
This is a nitpick but I was at first confused why we had contact DNS information

Reply:
Yeah, them's the breaks for language naming conventions...